aten/src/ATen/native/sparse/SparseUnaryOps.cpp

```diff
@@ -187,6 +192,28 @@ Tensor isinf_sparse_meta(const Tensor& self) {
   TORCH_CHECK_NOT_IMPLEMENTED(0, "nyi isinf for SparseMeta");
 }
 
+// Threshold_backward is not unary but it is the backward used for relu which is
+// unary
+Tensor threshold_backward_sparse(
+    const Tensor& grad_output,
+    const Tensor& self,
+    const Scalar& threshold) {
+  return coalesced_unary_ufunc(grad_output, [&](const Tensor& t) {
+    return at::threshold_backward(t, self.values(), threshold);
+  });
+}
+
+Tensor& threshold_backward_sparse_out(
+    const Tensor& grad_output,
+    const Tensor& self,
+    const Scalar& threshold,
+    Tensor& grad_input) {
+  return coalesced_unary_ufunc_out(
```

Comment:
@nikitaved I think you meant for your comment to go here? Self will be coalesced as it should be stashed for the `grad_fn` immediately after return. At least that is what the tests would suggest. 

In truth I am not sure, do we need to consider the possibility that one might call this function on its own, as in not as the gradient of another function?

Reply:
`self` is used in forward. What if I pass an uncoalesced `self` into RELU?

Reply:
This is a bit confusing the but the `self` parameter here to the is actually the result from `ReLU` which will always be coalesced. I used the `self` name to be consistent with the strided implementation. The relevant declaration in derivatives.yaml:

```yaml
- name: relu(Tensor self) -> Tensor
  self: threshold_backward(grad, result, 0)
  result: auto_element_wise
```

Although, if we were to implement `threshold` then `self` would actually be `self` and your point holds we may attempt to access values without coalescence. I will update this.

Reply:
Makes sense. Still `relu(uncoalesced input)` works just fine, right?